### ISSUE TITLE
Realm Sync - Realm cannot be opened and times out if local device time in future

### ISSUE THREAD

**Issue by AwildmanYozu** [realm/realm-js#3985]
### How frequently does the bug occur?

All the time

### Description

Expectation: If device time is in the future, I can open the realm.

Reality: If device time is in the future, opening the realm fails and times out.
The error logged is: [Error: undefined could not be downloaded in the allocated time: 60000 ms.]

Code to Replicate:

Realm.open(X)
      .then(realm => {
        console.log('Realm Resolved');
        realm.close();
        resolve('complete');
      })
      .catch(e => {
        console.log('Realm Error', e);
        reject(e);
      });
  });

Also tested same code with correct time and no issue occurs

### Can you reproduce the bug?

Yes, always

### Version

10.6.0

### What SDK flavour are you using?

MongoDB Realm (i.e. Sync, auth, functions)

### Are you using encryption?

Yes, using encryption

### Platform OS and version(s)

Android

### Build environment

Which debugger for React Native: ..

**Comment by kneth** [comment]
I am not able to reproduce it using node.js. 

@AwildmanYozu Do you only observe it on Android?

**Comment by fronck** [comment]
@AwildmanYozu A few questions to help us with reproduction:
Does the connection timeout immediately, or does it wait for the specified 60 seconds before reporting the timeout?
Also, does this problem happen when the device's clock is in the future before the app is launched, or only if the clock is adjusted while the app is running?

**Comment by josmanperez** [comment]
Hello @fronck, I manage to kind of reproduce this issue. I have a pretty simple function that opens a Realm with a certain partition. 

```   
const realm = await Realm.open({
    schema: [UserData],
    sync: {
        user: app.currentUser,
        partitionValue: "partition"
    }
})
.then(realm => {
    console.log('Realm Resolved');
    realm.close();
})
.catch(e => {
    console.log('Realm Error', e);
});
```

When executing this line of code in my laptop with the clock set on the right time. The logs shows something similar to this: 

```
Logged in with the user id: 615302500d5b65f2892a4c63
Connection[1]: Session[1]: client_reset_config = false, Realm exists = true, client reset = false
Connection[1]: Connected to endpoint '34.241.208.56:443' (from '192.168.1.58:49687')
Realm Resolved
Connection[1]: Disconnected
```

But If I change the clock to some time in the future, the output get stuck in this line (it doesn't matter if I wait, it will not go far this point)

```
Logged in with the user id: 615302500d5b65f2892a4c63
```
 
I am not getting the `[Error: undefined could not be downloaded in the allocated time: 60000 ms.` but the Realm is not opening for me either. 

I've tested this on NodeJS with 10.8.0

**Comment by kneth** [comment]
I can also reproduce it with a simple JS script (I don't see the error message but it might be that I wasn't patience) by manually setting my laptop's clock one hour in the future before running the script.

**Comment by josmanperez** [comment]
> I can also reproduce it with a simple JS script (I don't see the error message but it might be that I wasn't patient) by manually setting my laptop's clock one hour in the future before running the script.

@kneth that is exactly what I did. In my test, I waited more than 5 minutes but also did not get the message either. 

If you need any further tests, just let me know.

**Comment by kneth** [comment]
A fix is included in v10.10.1.
